features/grpc/exporter/src/main/java/org/opennms/features/grpc/exporter/GrpcHeaderInterceptor.java

```diff
@@ -29,19 +29,39 @@
 import io.grpc.ForwardingClientCall;
 import io.grpc.Metadata;
 import io.grpc.MethodDescriptor;
+import org.opennms.features.zenithconnect.persistence.api.ZenithConnectPersistenceException;
+import org.opennms.features.zenithconnect.persistence.api.ZenithConnectPersistenceService;
+import org.opennms.features.zenithconnect.persistence.api.ZenithConnectRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public class GrpcHeaderInterceptor implements ClientInterceptor {
+import java.util.Properties;
 
+public class GrpcHeaderInterceptor implements ClientInterceptor {
+    public static final Logger LOG = LoggerFactory.getLogger(GrpcHeaderInterceptor.class);
     private final Metadata metadata;
+    private ZenithConnectPersistenceService zenithConnectPersistenceService;
+    private final Metadata.Key AUTHORIZATION_KEY = Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
+    private final Metadata.Key TENANT_ID_KEY = Metadata.Key.of("tenant-id", Metadata.ASCII_STRING_MARSHALLER);
+    private final Metadata.Key AUTHORIZATION_BYPASS_KEY = Metadata.Key.of("Bypass-Authorization", Metadata.ASCII_STRING_MARSHALLER);
+    private final boolean ZENITH_CONNECT_ENABLED;
 
     public GrpcHeaderInterceptor(String tenantId) {
         metadata = new Metadata();
-        metadata.put(Metadata.Key.of("tenant-id", Metadata.ASCII_STRING_MARSHALLER), tenantId);
+        metadata.put(TENANT_ID_KEY, tenantId);
+        Properties onmsProperties = SystemUtils.getONMSSystemProperties();
+        ZENITH_CONNECT_ENABLED = Boolean.valueOf(onmsProperties.getProperty("opennms.zenithConnect.enabled", "false"));
```

Comment:
No, this should be part of configuration in the blueprint not as a system property.